src/app/child-dev-project/children/educational-material/educational-material-component/educational-material.component.ts

```diff
@@ -35,7 +46,8 @@ export class EducationalMaterialComponent implements OnInit {
     ],
   };
 
-  constructor(private entityMapper: EntityMapperService) {
+  constructor(private entityMapper: EntityMapperService,
+    private route: ActivatedRoute ) {
```

Comment:
To have maximum flexibility with our components, we do not load config from the route except for the top-most level (the full-screen views mostly implemented as `EntityDetailsComponent` or `EntityListComponent`). From there on, config objects are passed as `@Input` properties to the component. These are filled automatically by our [DynamicComponentDirective](https://github.com/Aam-Digital/ndb-core/blob/cc92709e6aa0fc10e230dda81b940e01fc804a9a/src/app/core/config/dynamic-components/dynamic-component.directive.ts#L51-L53).

Those background details aside, what that means for your implementation is:

1. define one or more inputs for your component here.  For your current implementation: `@Input() summary: any[]`
2. in the config (i.e. config-fix.ts for the sample), the configuration must be passed as a property "config" in the same object as the component name. For you that would be `{ "component": "EducationalMaterial", "config": { "summary": [/* your config */] } }`
3. The system is automatically assigning each property in the config object to the Input of the same name in your component.

I do realize we are missing any documentation on how to implement this. Will extend the developer docs following our discussion and will be requesting your feedback about that, then.

Reply:
@sleidig, I've made updates as per your suggestion. I've added a couple of test cases as well. Please review it.

Reply:
great :+1:   Thanks for writing tests also!